src/fprime/fpp/common.py

```diff
@@ -116,17 +129,23 @@ def execute(
 
         # Read files and arguments
         locations = self.get_locations_file(builder)
-        inputs = self.get_fpp_inputs(builder, context)
+        imports, sources = self.get_fpp_inputs(builder, context)
 
-        if not inputs:
-            print("[WARNING] No FPP inputs found in this module.")
+        if not sources:
+            print("[WARNING] No FPP sources found in this module.")
+
+        # Build the input argument list
+        input_args = []
+        if self.imports_as_sources:
+            input_args = [
+                str(item) for item in itertools.chain([locations] + imports + sources)
+            ]
+        else:
+            input_args.extend(["-i", ",".join(map(str, imports))] if imports else [])
+            input_args.extend(sources)
```

Comment:
Should this not be `[str(source) for source in sources]` to match the others?  Also, don't we need the `[locations]` bit?

Reply:
`subprocess.run()` does accept pathlib.Path objects, but googling it a little bit did show that this could lead to bugs every now and then (when debugging on specific versions of Windows? lol)... I'll change it for robustness's sake